### ISSUE TITLE
Flicker grid between map tiles on zoom

### ISSUE THREAD

**Issue by tobias-ullerich-init** [fleaflet/flutter_map#1156]
I have recently experienced a flicker grid between the map tiles when zooming on a map with rotation disabled. This bug seems to be depended on the Flutter SDK version and present since Flutter SDK 2.8.x. I tested this with the 0.14.0 version of the library, as well as the master branch. Running the application on Flutter 2.5.x will not lead to this bug. The bug will only appear with disabled map rotation interaction.
The attached video shows the graphical bug more precise. There is a line from north to south near London and a line from east to west at the height of the Label "United Kingdom".

https://user-images.githubusercontent.com/81169686/152992764-b0ee3241-c334-4eac-8f86-05600d7d2c5a.mp4

To reproduce the bug, simply use this code and run the application with the Flutter SDK 2.8.x or 2.10.x: 

```dart
FlutterMap(
  options: MapOptions(
    center: LatLng(51.5, -0.09),
    zoom: 5.0,
    interactiveFlags: InteractiveFlag.drag |
        InteractiveFlag.flingAnimation |
        InteractiveFlag.pinchMove |
        InteractiveFlag.pinchZoom |
        InteractiveFlag.doubleTapZoom,
  ),
  layers: [
    TileLayerOptions(
      urlTemplate:
          'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
      subdomains: ['a', 'b', 'c'],
      // For example purposes. It is recommended to use
      // TileProvider with a caching and retry strategy, like
      // NetworkTileProvider or CachedNetworkTileProvider
      tileProvider: NonCachingNetworkTileProvider(),
    ),
    MarkerLayerOptions(markers: markers)
  ],
)
```

**Comment by ibrierley** [comment]
This sounds similar to another problem, where you get a hairline between times sometimes. They found that if they removed the Opacity widget the hairlines were fixed.

I'm wondering if there's a change in Flutter where there's a rounding error or something on the Opacity widget (iirc there were some changes to Opacity in the latest flutter ver). It would be interesting if you get time to test removing it (apologies, I'll be away for over a week, so won't have time to do any debugging).

**Comment by JaffaKetchup** [comment]
Related to #1126 perhaps?

**Comment by tobias-ullerich-init** [comment]
> This sounds similar to another problem, where you get a hairline between times sometimes. They found that if they removed the Opacity widget the hairlines were fixed.

I removed the Opacity widget in the `_AnimatedTileState` and it worked, the hairlines are gone.

> Related to https://github.com/fleaflet/flutter_map/issues/1126 perhaps?

It seems to be the same issue I have.

**Comment by nxtSwitch** [comment]
As a package-side solution, we can use the `opacity` configuration from the `Image`, this solves the problem

<details>
  <summary>_AnimatedTileState.build</summary>

  ```dart
  @override
  Widget build(BuildContext context) {
    final tileWidget = (widget.tile.loadError && widget.errorImage != null)
        ? Image(
            fit: BoxFit.fill,
            image: widget.errorImage!,
            opacity: AlwaysStoppedAnimation(widget.tile.opacity),
          )
        : RawImage(
            fit: BoxFit.fill,
            image: widget.tile.imageInfo?.image,
            opacity: AlwaysStoppedAnimation(widget.tile.opacity),
          );

    return widget.tileBuilder == null
        ? tileWidget
        : widget.tileBuilder!(context, tileWidget, widget.tile);
  }
  ```
</details>

**Comment by ibrierley** [comment]
@tobias-ullerich-init I'm guessing you've just added your own solution you mentioned, but interested if you or anyone else has tried the PR mentioned https://github.com/fleaflet/flutter_map/pull/1169 as I think this issue is quite important.

**Comment by tranice** [comment]
I met the same issue.

**Comment by ibrierley** [comment]
Can you try a possible fix I added, https://github.com/fleaflet/flutter_map/pull/1169 see if it helps

**Comment by tobias-ullerich-init** [comment]
Hi, I was the last days on holidays but now I'm back at work and had some time to test your PR. You are right, as described above, I just removed the Opacity widget as a quick fix. But that wasn't a good solution for everyone. Now I tested your PR and the current released version. You PR looks good. I couldn't reproduce the issue anymore.

Thanks for helping and fixing this issue :)

**Comment by ibrierley** [comment]
Thanks for the feedback, thats really useful. Would be very interested for anyone else who specifically uses Opacity on their layer to try this.

**Comment by ibrierley** [comment]
Thanks, will see if I can get it reviewed and merged.

**Comment by JaffaKetchup** [comment]
Closing, as fixed by #1169.
